Utilities/Create-AzureContextFile.ps1

```diff
@@ -0,0 +1,14 @@
+# Remove the saved contexts from current PS session to avoid unauthorised export of saved contexts.
+Write-Host "Deleting the current Azure authorization information, if any."
```

Comment:
JSON  is human-readable, but we do not want unauthorized person to access. What about BSON?

Reply:
@juhlee-microsoft JSON is the default format in which Azure context files are saved. We don't have choice there. But, even though the JSON is human readable, it does not contain the user password. Instead, it contains Authenticated Session Key which is a long random string.

Reply:
JSON file contains subscription id, cached token and all keys. It's not comfortable situation putting all credential information in JSON format. You also addressed this concern in the first line.

Reply:
Let me elaborate this line - 
`Remove the saved contexts from current PS session to avoid unauthorized export of saved contexts`

This a precaution I'm taking in order to avoid the exporting Azure context file, if you are working from a Public Machine, or a guest user is operating from your machine.
Deleting the current PS context (not the .json file ) before and after creating an AzureContextFile makes sure that, your / guest credentials will be removed from the PowerShell session and Authentication Scenario 4 can not run, unless you explicitly provide a proper authentication.

The contents inside json file are totally managed by Azure Powershell cmdlets.
Now, we may convert JSON to BSON (not sure if that is possible) but still it doesn't serve the purpose you are asking for here. BSON is just encoded message (not the encrypted).
So, even if we convert it to BSON, it is not going to hide subscription ID and Username.

You can treat this json file same as our secret file (which is not encrypted and even though it has sensitive data). 
It is user's responsibility to make them secure. (Same rule applies to our secret file also).
So I don't think BSON is going to help here.

Reply:
Let me elaborate this line - 
`Remove the saved contexts from current PS session to avoid unauthorized export of saved contexts`

This a precaution I'm taking in order to avoid the exporting Azure context file, if you are working from a Public Machine, or a guest user is operating from your machine.
Deleting the current PS context (not the .json file ) before and after creating an AzureContextFile makes sure that, your / guest credentials will be removed from the PowerShell session and Authentication Scenario 4 can not run, unless you explicitly provide a proper authentication.

The contents inside json file are totally managed by Azure Powershell cmdlets.
Now, we may convert JSON to BSON (not sure if that is possible) but still it doesn't serve the purpose you are asking for here. BSON is just encoded message (not the encrypted).
So, even if we convert it to BSON, it is not going to hide subscription ID and Username.

You can treat this json file same as our secret file (which is not encrypted and even though it has sensitive data). 
It is user's responsibility to make them secure. (Same rule applies to our secret file also).
So I don't think BSON is going to help here.